### PR TITLE
warplda

### PR DESCRIPTION
**Pull request by Chris19920210**
你好，warplda 在实现的时候基本就参照的F+lda，不同体现在了sampler和trainer那部分，以及对partcsrresult有一定改进，现在的问题是init那部分过不去，increment那部分阻塞了。麻烦帮忙一起测下了，谢谢

### CODE REVIEW

**Comment by paynie** [comment]
OK，we will review  and test it

**Comment by leleyu** [comment]
@Chris19920210 hi, I have cloned your code and run the unit test.
I found that at line 253 of Trainer you wrote that
   ll += Gamma.logGamma(alpha + dk(value))
However, this should be 
   ll += Gamma.logGamma(alpha + value)
since that value is already the count number for topic assignment.

You can check this error at the stderr log of your worker node.
I think this might be the reason why your task will hang after initialization.

Unfortunately, there still be some other problems once this one is figured out.

File: angel-ps/mllib/src/main/java/com/tencent/angel/ml/warplda/Trainer.scala [hunk 1]
```diff
@@ -0,0 +1,500 @@
...
+        while (index.get() < n_docs) {
+          val d = index.incrementAndGet()
+          val dk = mutable.Map[Int,Int]()
+          if (d < n_docs) {
+            (data.accDoc(d) until data.accDoc(d+1)) foreach {j=>
+              val k = data.topics(data.inverseMatrix(j))
+              dk += k ->(dk.getOrElse(k, 0) + 1)
+            }
+            dk.foreach{case(k, value) =>
+              ll += Gamma.logGamma(alpha + dk(value))
```
Comment (leleyu) [inline angel-ps/mllib/src/main/java/com/tencent/angel/ml/warplda/Trainer.scala]: this should be  "ll += Gamma.logGamma(alpha + value)"

Comment (Chris19920210) [reply angel-ps/mllib/src/main/java/com/tencent/angel/ml/warplda/Trainer.scala]: Yeah, it's really a typo. Are your initialization get stucked as well? I'm even hanging during the during the initialization

**Comment by Chris19920210** [comment]
Hi, I found my errors for aliassampling for generating the proposals, just commit to my fork

**Comment by Chris19920210** [comment]
Hi, I correct two bugs and commit to my fork. Sorry for that

**Pull request merged**